tsdb/shard.go

```diff
@@ -150,24 +150,22 @@ func NewShard(id uint64, index *DatabaseIndex, path string, walPath string, opti
 		database:        db,
 		retentionPolicy: rp,
 
-		logger:       log.New(os.Stderr, "[shard] ", log.LstdFlags),
-		logOutput:    os.Stderr,
+		logger:       logger,
+		baseLogger:   logger,
 		EnableOnOpen: true,
 	}
 	return s
 }
 
-// SetLogOutput sets the writer to which log output will be written. It is safe
-// for concurrent use.
-func (s *Shard) SetLogOutput(w io.Writer) {
-	s.logger.SetOutput(w)
+func (s *Shard) WithLogger(log zap.Logger) {
+	s.baseLogger = log.With(
```

Comment:
This seems like only half of the change to switch to a full structured log. Without changing the current message being printed later on, it results in long messages with duplicated information:
```
[I] 2016-12-14T16:25:12Z /home/jlegasse/.influxdb/data/_internal/monitor/9 database index loaded in 8.860841ms shard=9 path=/home/jlegasse/.influxdb/data/_internal/monitor/9 service=shard
```

I think for this change, the extra context should be left out. We can revisit later and switch from formatted strings to zap context values all at once, then.